modules/tide_edit_protection/tide_edit_protection.module

```diff
@@ -142,3 +143,38 @@ function tide_edit_protection_preprocess_status_messages(&$variables) {
     }
   }
 }
+
+/**
+ * Implements hook_entity_predelete().
+ */
+function tide_edit_protection_entity_predelete(Drupal\Core\Entity\EntityInterface $entity) {
+  if (PHP_SAPI == 'cli') {
+    $entity_id = $entity->id();
+    $entity_type = $entity->getEntityTypeId();
+    $lock_service = \Drupal::service('content_lock');
+    if (!$lock_service->isLockable($entity)) {
+      return;
+    }
+    $data = $lock_service->fetchLock($entity_id, NULL, $entity->language()
+      ->getId(), $entity_type);
+    if ($data !== FALSE) {
+      $lock_service->release($entity_id, LanguageInterface::LANGCODE_NOT_SPECIFIED);
+    }
+  }
+}
+
+/**
+ * Implements hook_module_implements_alter().
+ */
+function tide_edit_protection_module_implements_alter(&$implementations, $hook) {
+  if (PHP_SAPI == 'cli') {
```

Comment:
@vincent-gao just increase module_weight of `content_lock` to a big number (eg. 1000) using an update hook in tide_edit_protection. No need to implement this hook.